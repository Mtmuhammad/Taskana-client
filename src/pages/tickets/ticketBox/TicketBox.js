// ticket box component for ticket page
import React from 'react' 

const TicketBox = ({ setShowTickets, name, tickets, color }) => {
  return (
    <div className="col-md-6 col-lg-3">
      <div
      onClick={() => setShowTickets(tickets)}
        data-testid="ticket-box"
        style={{
          borderColor: `${color}20`,
          color: `${color}`,
          backgroundColor: `${color}20`,
        }}
        className="mb-4 ticketBox card"
      >
        <div
          data-testid="ticket-box-body"
          className="px-5 py-3 rounded text-center"
        >
          <h1 data-testid="ticket-box-number">{tickets?.length}</h1>
          <h6 data-testid="ticket-box-title">{name}</h6>
        </div>
      </div>
    </div>
  );
};

export default TicketBox;
