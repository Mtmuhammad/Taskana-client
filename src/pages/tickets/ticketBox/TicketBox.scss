@use "../../../styles/abstracts/_variables" as *;
@use "../../../styles/abstracts/_mixins" as *;


.tickets {
   & .card {
      background-color: var(--primary-color-light);
      color: var(--text-color);
   }
}

.ticketBox {
   & h1 {
      font-size: 4.5rem;
   }

   & h6 {
      font-size: 1.8rem;
   }
}