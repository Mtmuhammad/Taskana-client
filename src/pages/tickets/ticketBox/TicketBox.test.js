import React from "react";
import { render, screen } from "@testing-library/react";
import TicketBox from "./TicketBox";
import { BrowserRouter } from "react-router-dom";

describe("smoke and snapshot tests", () => {
  test("should render without crashing", () => {
    render(<TicketBox />, { wrapper: BrowserRouter });
  });

  test("should match snapshot", () => {
    const { asFragment } = render(<TicketBox />, { wrapper: BrowserRouter });
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("ticket box", () => {
  test("should render ticket box", () => {
    render(<TicketBox />, { wrapper: BrowserRouter });
    const container = screen.getByTestId("ticket-box");
    expect(container).toBeInTheDocument();
    expect(container).toHaveClass("mb-4 ticketBox card");
  });
  test("should render ticket box body", () => {
    render(<TicketBox />, { wrapper: BrowserRouter });
    const body = screen.getByTestId("ticket-box-body");
    expect(body).toBeInTheDocument();
    expect(body).toHaveClass("px-5 py-3 rounded text-center");
  });
  test("should render ticket box number", () => {
    render(<TicketBox />, { wrapper: BrowserRouter });
    const number = screen.getByTestId("ticket-box-number");
    expect(number).toBeInTheDocument();
  });
  test("should render ticket box title", () => {
    render(<TicketBox />, { wrapper: BrowserRouter });
    const number = screen.getByTestId("ticket-box-title");
    expect(number).toBeInTheDocument();
  });
});
