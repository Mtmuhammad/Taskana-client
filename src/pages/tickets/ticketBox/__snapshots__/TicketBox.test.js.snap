// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="col-md-6 col-lg-3"
  >
    <div
      class="mb-4 ticketBox card"
      data-testid="ticket-box"
    >
      <div
        class="px-5 py-3 rounded text-center"
        data-testid="ticket-box-body"
      >
        <h1
          data-testid="ticket-box-number"
        />
        <h6
          data-testid="ticket-box-title"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
